src/agent/testrunpublisher.ts

```diff
@@ -164,24 +192,80 @@ export class TestRunPublisher {
     //-----------------------------------------------------
     public publishTestRun(resultFilePath: string): Q.Promise<testifm.TestRun> {
         var defer = Q.defer();
-        
+
         var _this = this;
         var testRunId;
-        var results; 
+        var results;
 
-        _this.readResults(resultFilePath).then(function (res) {
+        _this.readResults(resultFilePath).then(function(res) {
             results = res.testResults;
             return _this.startTestRun(res.testRun);
-        }).then(function (res) {
+        }).then(function(res) {
             testRunId = res.id;
             return _this.addResults(testRunId, results);
-        }).then(function (res) {
+        }).then(function(res) {
             return _this.endTestRun(testRunId, resultFilePath);
-        }).then(function (res) {
+        }).then(function(res) {
             defer.resolve(res);
-        }).fail(function (err) {
+        }).fail(function(err) {
             defer.reject(err);
-        }); 
+        });
+
+        return defer.promise;
+    }
+
+    //-----------------------------------------------------
+    // Publish a test run
+    // - resultFiles: string - Path to the results files
+    //-----------------------------------------------------
+    public publishMergedTestRun(resultFiles: string[]): Q.Promise<testifm.TestRun> {
+        var defer = Q.defer();
+
+        var _this = this;
+        var testRunId;
+
+        var totalTestCaseDuration: number = 0;
+        var totalTestResults: testifm.TestResultCreateModel[] = [];
+        var testName;
+        var currentTime = Date.now();
+
+        for (var i = 0; i < resultFiles.length; i++) {
+            var report = _this.readResults(resultFiles[i]).then(function(res) {
+                res.testResults.forEach(tr => {
+                    totalTestCaseDuration += +tr.durationInMs;
+                });
+                totalTestResults = totalTestResults.concat(res.testResults);
+            });
+
+        }
+
+        var startDate = new Date(currentTime);
+        var completedDate = new Date(currentTime + totalTestCaseDuration);
+
+        //create test run data
+        var testRun = <testifm.RunCreateModel>{
+            name: this.runContext.runTitle,
+            startDate: startDate.toISOString(),
+            completeDate: completedDate.toISOString(),
+            state: "InProgress",
+            automated: true,
+            buildPlatform: this.runContext.platform,
+            buildFlavor: this.runContext.config,
+            build: { id: this.runContext.buildId },
+            releaseUri: this.runContext.releaseUri,
+            releaseEnvironmentUri: this.runContext.releaseEnvironmentUri
+        };
+
+        _this.startTestRun(testRun).then(function(res) {
+            testRunId = res.id;
+            return _this.addResults(testRunId, totalTestResults);
+        }).then(function(res) {
```

Comment:
If anything fails here, the test run probably remains  in progress state.
Why don't we mark the run aborted on failure?